configs/build_config.py

```diff
@@ -127,15 +126,23 @@ def __init__(self, flask_config_path, autoratify=True):
                 raise ValueError('\n'.join(errors))
 
 
-    def get(self, key, default_value):
+    def get(self, key, default_value=None):
         ''' Same functionality as of dict.get(). Return a key if in the settings,
         or default_value instead. '''
+        if key == 'GOLDEN_IMAGE': #golden name is dynamically calculated
+            return self.golden_image
+
         return self._settings.get(key, default_value)
 
+    def update(self, new_dict):
+        self._settings.update(new_dict)
 
     # Duck-type a dict.  It's empty before extract_flask_config().  The
     # flask routines turn the string value 'None' into Python None.
     def __getitem__(self, key):
+        if key == 'GOLDEN_IMAGE': #golden name is dynamically calculated
+            return self.golden_image
+
         return self._settings.get(key)
```

Comment:
Is this why there's a get?   There can't be a default here, it's part of the descriptor protocol (aka data model, see https://docs.python.org/3/reference/datamodel.html#object.__getattr__)   and could directly access self.__dict__

For a default of "None" for missing attributes, wrap this in a try-except KeyError, then return None in the except clause.

Reply:
then put "some/default/path" somewhere else than carrying around a string literal in the middle of code where it's hard to find.  Or, look how I handled all the default values in one place at

https://github.com/coloroco/FAME-Z/blob/master/ivshmem_twisted/ivshmem_twisted.py#L255